src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -564,6 +564,9 @@ internal static string GetPipeName(
             // of this method.
             clientDirectory = clientDirectory.TrimEnd(Path.DirectorySeparatorChar);
 
+            // Similarly, we don't want multiple servers if the path differs in casing.
```

Comment:
Is that applicable only on Windows?

Reply:
Any OS can have case insensitive paths, or even paths for which parts have case insensitivity.